### PR TITLE
Override Notion Client logger

### PR DESCRIPTION
**Pull request by PopDaph**
Trying to prevent NotionClient to spam our datadog: https://app.datadoghq.eu/logs?query=-status%3A%28warn%20OR%20info%29&cols=host%2Cservice&index=%2A&messageDisplay=inline&refresh_mode=sliding&stream_sort=desc&viz=stream&from_ts=1700649104874&to_ts=1700650004874&live=true

### CODE REVIEW

File: connectors/src/connectors/notion/lib/notion_api.ts [hunk 1]
```diff
@@ -30,6 +31,14 @@ import mainLogger from "@connectors/logger/logger";
 
 const logger = mainLogger.child({ provider: "notion" });
 
+const notionClientLogger = (
+  level: LogLevel,
+  message: string,
+  extraInfo: Record<string, unknown>
+) => {
+  console.log(`[${level}]: ${message}`, extraInfo);
```
Comment (spolu) [inline connectors/src/connectors/notion/lib/notion_api.ts:39]: Why not directly use a logger here? potentially deactivating after that?

Comment (fontanierh) [reply connectors/src/connectors/notion/lib/notion_api.ts:39]: We def don't want those logs to clutter our DD errors no ? The'll handle the requests to notion, and we'll do the logging if where we need it wdyt ?

Comment (spolu) [reply connectors/src/connectors/notion/lib/notion_api.ts:39]: I'm not too worried about cluttering. My point is that console.log will clutter more than logger ?

Comment (PopDaph) [reply connectors/src/connectors/notion/lib/notion_api.ts]: I removed the console.log and no-op instead > want me to add back a log in info?

Comment (PopDaph) [reply connectors/src/connectors/notion/lib/notion_api.ts]: OK added back with a logger.info so we keep them but stop being spammed with errors

**Review by spolu — APPROVED** [review]

**Pull request merged**